manubot/cite/curie/__init__.py

```diff
@@ -104,18 +104,22 @@ def _download_bioregistry():
     response.raise_for_status()
     results = response.json()
     assert isinstance(results, dict)
-    for prefix, metadata in results.items():
-        assert isinstance(metadata, dict)
-        for field in set(metadata) - _keep_bioregistry_fields:
-            del metadata[field]
-        metadata["prefix"] = prefix
-        metadata["all_prefixes"] = sorted(
+    registry = list()
+    for prefix, resource in results.items():
+        assert isinstance(resource, dict)
+        if not resource.get("uri_format"):
+            # discard unresolvable prefixes
+            continue
+        for field in set(resource) - _keep_bioregistry_fields:
+            del resource[field]
+        resource["prefix"] = prefix
+        resource["all_prefixes"] = sorted(
             {
                 prefix,
-                *(x.lower() for x in metadata.get("synonyms", [])),
+                *(x.lower() for x in resource.pop("synonyms", [])),
             }
         )
-    registry = list(results.values())
+        registry.append(resource)
     json_text = json.dumps(registry, indent=2, ensure_ascii=False)
```

Comment:
why not just do `sort_keys=True` in `json.dumps` to get more determinism?

Reply:
The non-determinism was with values of `synonyms`, so `sort_keys` would not fix it. Also python dicts are now deterministic so sort_keys=False doesn't introduce non-determinism in field order.